crates/evm/src/block.rs

```diff
@@ -344,90 +344,76 @@ where
+            .map_err(BlockExecutionError::other)?;
+
+        // Get existing account info or create default
+        let mut acc_info = acc.account_info().unwrap_or_default();
+
+        // Only initialize if the account has no code
+        if acc_info.is_empty_code_hash() {
+            // Set the keychain code
+            let code = Bytecode::new_legacy(Bytes::from_static(&[0xef]));
+            acc_info.code_hash = code.hash_slow();
+            acc_info.code = Some(code);
+
+            // Convert to revm account and mark as touched
+            let mut revm_acc: Account = acc_info.into();
+            revm_acc.mark_touch();
+
+            // Commit the account to the database to ensure it persists
+            // even if no transactions are executed in this block
+            db.commit(HashMap::from_iter([(ACCOUNT_KEYCHAIN_ADDRESS, revm_acc)]));
         }
 
-        // Modify CreateX bytecode and upgrade Multicall to Multicall3 if AllegroModerato is active
-        if self
-            .inner
-            .spec
-            .is_allegro_moderato_active_at_timestamp(block_timestamp)
-        {
-            let evm = self.evm_mut();
-            let db = evm.ctx_mut().db_mut();
-
-            // Update CreateX bytecode if outdated
-            let acc = db
-                .load_cache_account(CREATEX_ADDRESS)
-                .map_err(BlockExecutionError::other)?;
-
-            let mut acc_info = acc.account_info().unwrap_or_default();
-
-            let correct_code_hash =
-                tempo_contracts::contracts::CREATEX_POST_ALLEGRO_MODERATO_BYTECODE_HASH;
-            if acc_info.code_hash != correct_code_hash {
-                acc_info.code_hash = correct_code_hash;
-                acc_info.code = Some(Bytecode::new_legacy(
-                    tempo_contracts::contracts::CREATEX_POST_ALLEGRO_MODERATO_BYTECODE,
-                ));
+        // Modify CreateX bytecode and upgrade Multicall to Multicall3
+        let evm = self.evm_mut();
+        let db = evm.ctx_mut().db_mut();
 
-                let mut revm_acc: Account = acc_info.into();
-                revm_acc.mark_touch();
+        // Update CreateX bytecode if outdated
+        let acc = db
+            .load_cache_account(CREATEX_ADDRESS)
+            .map_err(BlockExecutionError::other)?;
 
-                db.commit(HashMap::from_iter([(CREATEX_ADDRESS, revm_acc)]));
-            }
+        let mut acc_info = acc.account_info().unwrap_or_default();
 
-            // Update Multicall to Multicall3 if outdated
-            let acc = db
-                .load_cache_account(MULTICALL_ADDRESS)
-                .map_err(BlockExecutionError::other)?;
+        let correct_code_hash = tempo_contracts::contracts::CREATEX_BYTECODE_HASH;
+        if acc_info.code_hash != correct_code_hash {
+            acc_info.code_hash = correct_code_hash;
+            acc_info.code = Some(Bytecode::new_legacy(
+                tempo_contracts::contracts::CREATEX_BYTECODE,
+            ));
```

Comment:
I believe we can remove this entirely? The correct bytecode should be deployed at genesis.